ZenPacks/zenoss/CalculatedPerformance/dsplugins.py

```diff
@@ -174,10 +188,16 @@ def collect(self, config, datasource, rrdcache, collectionTime):
             collectedValues[datasource.component]['_'.join((datasource.datasource, datapoint.id))] = \
                 (aggregate, collectionTime)
 
-        returnValue({
+        data = {
             'events': collectedEvents,
-            'values': collectedValues,
-        })
+            'values': collectedValues,}
+
+        if useBasisInterval and len(firstTarget) > 1:
```

Comment:
It seems dangerous to assume that the difference between two samples should be the interval. Even in cases where network connectivity to the target is lost for a while you will end up with a very length time before the calculated datasource will be collected again.

There are probably a lot of mechanisms you could use to mitigate these kinds of problems. The most direct might be to add two more configuration options to the datasource types that allow the user to specify a minimum and maximum interval when useBasisInterval is enabled.